mantas.js

```diff
@@ -0,0 +1,57 @@
+import { get } from "request";
+import { writeFile } from "fs";
+
+const DAYS_IN_WEEK = 7;
+const daysInMonth = 30;
+
+const songs = ["Back In Black", "Stairway to Heaven", "Hey Jude"];
+const Artists = ["ACDC", "Led Zeppelin", "The Beatles"];
+
+function eraseDatabase() {}
+function restore_database() {}
+
+class animal {}
+class Alpaca {}
+
+const Car = {
+  carMake: "Honda",
+  carModel: "Accord",
+  carColor: "Blue"
+};
+
+function hashIt(data) {
+    // The hash
+    let hash = 0;
+
+    // Length of string
+    const length = data.length;
+
+    // Loop through every character in data
+    for (let i = 0; i < length; i++) {
+        // Get character code.
+        const char = data.charCodeAt(i);
+        // Make the hash
+        hash = (hash << 5) - hash + char;
+        // Convert to 32-bit integer
+        hash &= hash;
+    }
+}
+
+get(
+    "https://en.wikipedia.org/wiki/Robert_Cecil_Martin",
+    (requestErr, response, body) => {
+        if (requestErr) {
+            console.error(requestErr);
+        } else {
+            writeFile("article.html", body, writeErr => {
+                if (writeErr) {
+                    console.error(writeErr);
+                } else {
+                    console.log("File written");
+                }
+            });
+        }
+    }
+);
```

Comment:
Atgaliniai skambučiai (angl. callbacks) nėra švarūs ir sukelia pernelyg didelę tabuliaciją (angl. nesting). Geriau naudoti pažadus (angl. promises).

```suggestion
get("https://en.wikipedia.org/wiki/Robert_Cecil_Martin")
    .then(body => {
        return writeFile("article.html", body);
    })
    .then(() => {
        console.log("File written");
    })
    .catch(err => {
        console.error(err);
    });
```

Pažadai yra švari alternatyva atgaliniams skambučiams, tačiau asinchronizavimas ir laukimas siūlo dar švaresnį sprendimą.

```suggestion
async function getCleanCodeArticle() {
    try {
        const body = await get("https://en.wikipedia.org/wiki/Robert_Cecil_Martin");
        await writeFile("article.html", body);
        console.log("File written");
    } catch (err) {
        console.error(err);
    }
}

getCleanCodeArticle();
```